modules/hdf/include/opencv2/hdf/hdf5.hpp

```diff
@@ -103,13 +103,109 @@ class CV_EXPORTS_W HDF5
      */
     CV_WRAP virtual bool hlexists( const String& label ) const = 0;
 
-    /* @overload */
+    /**
+     * Check whether a given attribute exits or not in the root group.
+     *
+     * @param atlabel the attribute name to be checked.
+     * @return true if the attribute exists, false otherwise.
+     *
+     * @sa atdelete, atwrite, atread
+     */
+    CV_WRAP virtual bool atexists(const String& atlabel) const = 0;
+
+    /**
+     * Delete an attribute from the root group.
+     *
+     * @param atlabel the attribute to be deleted.
+     *
+     * @note CV_Error() is called if the given attribute does not exist. Use atexists()
+     * to check whether it exists or not beforehand.
+     *
+     * @sa atexists, atwrite, atread
+     */
+    CV_WRAP virtual void atdelete(const String& atlabel) = 0;
+
+    /**
+     * Write an attribute inside the root group.
+     *
+     * @param value attribute value.
+     * @param atlabel attribute name.
+     *
+     * The following example demonstrates how to write an attribute of type cv::String:
+     *
+     *  @snippet samples/read_write_attributes.cpp snippets_write_str
+     *
+     * @note CV_Error() is called if the given attribute already exists. Use atexists()
+     * to check whether it exists or not beforehand. And use atdelete() to delete
+     * it if it already exists.
+     *
+     * @sa atexists, atdelete, atread
+     */
+    CV_WRAP virtual void atwrite(const int value, const String& atlabel) = 0;
```

Comment:
BTW, Current OpenCV interfaces usually pass required "label" before "value":
- imshow(window_name, mat)
- imwrite(file_name, img)
- [FileStorage](https://github.com/opencv/opencv/blob/3.4.0/modules/core/include/opencv2/core/persistence.hpp#L695) utility functions

Reply:
Yes, you are right. I implemented it originally like the one you suggested.
But I found that other member functions in the same file put the label after the value,
for example, 
 - [dswrite](https://github.com/opencv/opencv_contrib/blob/master/modules/hdf/include/opencv2/hdf/hdf5.hpp#L298)
 - [dsread](https://github.com/opencv/opencv_contrib/blob/master/modules/hdf/include/opencv2/hdf/hdf5.hpp#L432)

To conform the style used in the same file, I put the label after the value.